### PR TITLE
Scripts/Stratholme: Convert some Bosses to EventMaps

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**

### CODE REVIEW

**Comment by xerkoss** [comment]
why not BossAI? :D

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp [hunk 1]
```diff
@@ -116,6 +124,12 @@ class boss_baron_rivendare : public CreatureScript
                 summoned->AI()->AttackStart(target);
         }
 
+        void DoSpawnSkeleton()
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: There are spells to do this

File: sql/updates/world/2015_01_10_00_world.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+(17478, 0, 329, 4048.87, -3363.22, 115.05, 3.62, 0),
+(17479, 0, 329, 4051.77, -3350.89, 115.05, 3.06, 0),
+(17480, 0, 329, 4048.37, -3339.96, 115.05, 2.45, 0);
+
+-- Mindless Skeleton SAI
+SET @ENTRY := 11197;
+UPDATE `creature_template` SET `AIName`="SmartAI", `ScriptName`='' WHERE `entry`=@ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(@ENTRY,0,0,0,54,0,100,0,0,0,0,0,49,0,0,0,0,0,0,21,50,0,0,0,0,0,0,"Mindless Skeleton - On Just Summoned - Start Attacking");
```
Comment (joschiwald) [inline sql/updates/world/2015_01_10_00_world.sql]: sai is unneeded, handled in boss script

Comment (Rushor) [inline sql/updates/world/2015_01_10_00_world.sql]: JustSummoned does not worked for me in this case. without the sai, some of them only stand around and do nothing

Comment (jackpoz) [inline sql/updates/world/2015_01_10_00_world.sql]: what doesn't work ? is there a bug that needs to be fixed (and is the SAI part a workaround ? ) ?

Comment (Rushor) [inline sql/updates/world/2015_01_10_00_world.sql]: I was wrong, dunno why, but now it works for me even without the sai - strange

**Comment by MitchesD** [comment]
what about commented parts? they have been commented for some reason, right, but you could try them and maybe fix them ;-)

btw, squash your commits pls

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp [hunk 3]
```diff
@@ -116,6 +113,16 @@ class boss_baron_rivendare : public CreatureScript
...
+        {
+            DoCast(me, SPELL_RAISE_DEAD1);
```
Comment (Gooyeth) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: It is not better?, could even drop the function DoSpawnSkeleton()
```c++
for (uint32 i = SPELL_RAISE_DEAD1 <= SPELL_RAISE_DEAD6; ++i)
     DoCast(me, i);
```

**Comment by Rushor** [comment]
okay yes i will try the part with the heal of the skeletons, do you have a short guide (especially for tc) to squash the three commits into one?

**Comment by Aokromes** [comment]
http://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp [hunk 4]
```diff
@@ -53,13 +53,19 @@ enum Spells
...
-// Define Add positions
-Position const ADD_POS_1 = {4017.403809f, -3339.703369f, 115.057655f, 5.487860f};
-Position const ADD_POS_2 = {4013.189209f, -3351.808350f, 115.052254f, 0.134280f};
-Position const ADD_POS_3 = {4017.738037f, -3363.478016f, 115.057274f, 0.723313f};
-Position const ADD_POS_4 = {4048.877197f, -3363.223633f, 115.054253f, 3.627735f};
-Position const ADD_POS_5 = {4051.777588f, -3350.893311f, 115.055351f, 3.066176f};
-Position const ADD_POS_6 = {4048.375977f, -3339.966309f, 115.055222f, 2.457497f};
+enum RivendareMisc
+{
+    EVENT_SHADOW_BOLT           = 0,
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: Zero isn't a valid index for events, events scheduled with 0 as id are never executed

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp [hunk 5]
```diff
@@ -126,53 +133,42 @@ class boss_baron_rivendare : public CreatureScript
...
+                        break;
+                    case EVENT_CLEAVE:
+                        DoCastVictim(SPELL_CLEAVE);
+                        _events.ScheduleEvent(EVENT_CLEAVE, urand(7, 17) * IN_MILLISECONDS);
+                        break;
+                    case EVENT_MORTAL_STRIKE:
+                        DoCastVictim(SPELL_MORTALSTRIKE);
+                        _events.ScheduleEvent(EVENT_MORTAL_STRIKE, urand(10, 25) * IN_MILLISECONDS);
+                        break;
+                        /*case EVENT_RAISE_DEAD:
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: If I'm not mistaken this spell has TARGET_SRC_AREA_ENTRY flags and all it needs to work is a target condition (it should hit the skeletons as it seems). Raise dead is in fact death pact... that's all I remember from my memories

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp [hunk 6]
```diff
@@ -35,6 +35,14 @@ enum Spells
     //SPELL_POSSESS           = 17244
 };
 
+enum AnastariSpells
+{
+    EVENT_BANSHEEWAIL       = 0,
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baroness_anastari.cpp]: Same thing here, event indexed as 0

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp (hunk 4 above)
Comment (jackpoz) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: was this PR tested being being submitted for review ?

File: src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp (hunk 5 above)
Comment (Shauren) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp]: The reason this doesnt work is this - SPELL_RAISEDEAD = 15708 // bad copypaste of value (15708 is mortal strike)

**Comment by jackpoz** [comment]
this convertion PR actually breaks the event due to errors in how EventMap events are handled, in particular 0 used as event id. this means the code hasn't been tested and/or debugged, please submit a pull request only when you are sure it works, otherwise state it clearly in the PR description that it is work-in-progress untested code

**Pull request closed without merging**